### PR TITLE
refactor account id to be in request header

### PR DESCRIPTION
**Pull request by alexjoybc**
# Description

This PR includes the following proposed change(s):

- [FLA-180](https://justice.gov.bc.ca/jira/browse/FLA-180) - refactor account id to be in request header
- fix demo accounts return
- add missing unit tests

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/libs/efiling-account-client/src/main/java/ca/bc/gov/open/jag/efilingaccountclient/CsoAccountServiceImpl.java [hunk 1]
```diff
@@ -72,9 +65,9 @@ private AccountDetails getCsoDetails(String userGuid)  {
         try {
             profiles.addAll(accountFacadeBean.findProfiles(userGuid));
         } catch (NestedEjbException_Exception e) {
-            throw new EfilingAccountServiceException("Exception while fetching account details", e.getCause());
+            throw new EfilingAccountServiceException("Exception while fetching account details", e);
         }
-        //An account must only one profile associated to proceed
+        // An account must only one profile associated to proceed
```
Comment (sdevalapurkar-bcgov) [inline src/backend/libs/efiling-account-client/src/main/java/ca/bc/gov/open/jag/efilingaccountclient/CsoAccountServiceImpl.java:70]:
```suggestion
        // An account must have only one profile associated with it to proceed
```

File: src/backend/libs/efiling-account-client/src/test/java/ca/bc/gov/open/jag/efilingaccountclient/CsoAccountServiceImplTest.java [hunk 2]
```diff
@@ -19,15 +21,21 @@
...
 public class CsoAccountServiceImplTest {
 
-    private static final String USERGUIDNOROLE = "USERGUIDNOROLE";
-    private static final String USERGUIDWITHFILEROLE = "USERGUIDWITHFILEROLE";
-    private static final String USERGUIDWITHNOCSO = "USERGUIDWITHNOCSO";
+    private static final UUID USER_GUID_NO_ROLE = UUID.randomUUID();
+    private static final UUID USER_GUID_WITH_FILE_ROLE = UUID.randomUUID();
+    private static final UUID USER_GUID_WITH_NO_CSO = UUID.randomUUID();
+    private static final UUID USER_GUID_WITH_EJB_EXCEPTION = UUID.randomUUID();
+    private static final UUID USER_GUID_WITH_MULTIPROFILE = UUID.randomUUID();
```
Comment (sdevalapurkar-bcgov) [inline src/backend/libs/efiling-account-client/src/test/java/ca/bc/gov/open/jag/efilingaccountclient/CsoAccountServiceImplTest.java:35]:
```suggestion
    private static final UUID USER_GUID_WITH_MULTI_PROFILE = UUID.randomUUID();
```

File: src/frontend/efiling-demo/src/components/page/home/Home.js [hunk 3]
```diff
@@ -42,10 +42,14 @@ const input = {
...
-
   axios
-    .post(`/submission/generateUrl`, updatedUrlBody)
+    .post(
+      `/submission/generateUrl`,
+      {
+        headers: { "X-Auth-UserId": accountGuid }
+      },
+      urlBody
+    )
```
Comment (sdevalapurkar-bcgov) [inline src/frontend/efiling-demo/src/components/page/home/Home.js:52]: nice! thanks

File: README.md [hunk 4]
```diff
@@ -44,7 +44,7 @@ Generic File Submission API (to be used by the Family Law Act Application at fir
 ## Running the App
 
 By default a demo mode is enabled.
-if you want to integrate with the CSO application create a local `.env` at the root of the repository based off `.env.template`. 
+if you want to integrate with the CSO application create a local `.env` at the root of the repository based off `.env.template`.
```
Comment (akroon3r) [inline README.md:47]: ...with the CSO application, first create...

suggestion, non-blocking

Comment (alexjoybc) [inline README.md:47]:
```suggestion
if you want to integrate with the CSO application, first create a local `.env` at the root of the repository based off `.env.template`.
```

Comment (alexjoybc) [reply README.md]: done ;-)

**Pull request merged**